lib/webmock/fixtures/manager.rb

```diff
@@ -3,6 +3,108 @@ module WebMock
+        @started_fixtures = {}
+      end
+
+      # Fetch the `WebMock::RequestStub` for any started fixtures
+      # @param fixture_name [Symbol] the symbol name of the fixture
+      # @return [WebMock::RequestStub] the started fixture stub
+      # @raise [KeyError] if the fixture requested has not been started
+      def [](fixture_name)
+        # DEV: `fetch` wil raise `KeyError` if `fixture_name` does not exist
+        return @started_fixtures.fetch(fixture_name)
+      end
+
+      # Store an instance of `WebMock::RequestStub` for a given fixture
+      # @param fixture_name [Symbol] the symbol name of the fixture started
+      # @param stub [WebMock::RequestStub] the stub that was started for the fixture
+      def []=(fixture_name, stub)
+        @started_fixtures[fixture_name] = stub
+      end
+
+      # Ensure every subclass of `Manager` sets `@registered_fixtures` to `{}` (by default it will be `nil`)
+      # @classmethod
+      # @param subclass [Class] the class instance which is subclassing `Manager`
+      def self.inherited(subclass)
+        subclass.instance_variable_set(:@registered_fixtures, {})
+      end
+
+      # Preload a web fixture which can then be started via `Manager.run`
+      # @classmethod
+      # @param name [Symbol] the name to register the fixture as
+      # @param verb [Symbol] symbol method to mock (e.g. `:get`, `:post`)
+      # @param pattern [Regexp|String] URI pattern to match for this mock
+      # @param response [String|File|Lambda|Hash] the response, this is the same as what would be supplied
+      #   to `WebMock::RequestStub#to_return` please see https://github.com/bblimke/webmock for examples
+      def self.register_fixture(name, verb, pattern, response)
+        @registered_fixtures[name] = {
+          :pattern => pattern,
+          :verb => verb,
+          :response => response,
+        }
+      end
+
+      # Retrieve a hash of registered fixtures available to this Manager
+      # @classmethod
+      # @return [Hash] a hash mapping the registered name to a hash of properties
+      #   e.g. `{ :get_example => { :pattern => %r{example.org}, :verb => :get, :response => "Hello World" } }`
+      def self.registered_fixtures
+        return @registered_fixtures
+      end
+
+      # Reset this manager by removing all previously registered fixtures
+      # @classmethod
+      def self.reset!
+        @registered_fixtures = {}
+      end
+
+      # Preload a web fixture from a file which can then be started via `Manager.run`
+      # @classmethod
+      # @param name [Symbol] the name to register the fixture as
+      # @param verb [Symbol] symbol method to mock (e.g. `:get`, `:post`)
+      # @param pattern [Regexp|String] URI pattern to match for this mock
+      # @param file_name [String] the name of the file to load the fixture response from
+      def self.load_fixture_file(name, verb, pattern, file_name)
```

Comment:
Thoughts on renaming this to `register_fixture_file`? There's nothing in Ruby that says "load" when talking about files